### PR TITLE
WIP - Notes

### PR DESCRIPTION
**Pull request by ewilde**
Signed-off-by: Edward Wilde <ewilde@gmail.com>

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Installing OpenFaaS Cloud requires some initial setup with GitHub or GitLab and
 * Kubernetes or Docker Swarm (Docker cannot be running with XFS as a backing file-system due to buildkit restrictions)
 * Registry account - Docker Hub account or private registry with TLS
 * OpenFaaS deployed with authentication enabled
-* Extended timeouts for the queue-worker, gateway and the backend provider
```
Comment (ewilde) [inline docs/README.md]: What are the recommended timeouts? Can we give some pointers here? example config etc... I only see something for the queue-worker 👇

File: docs/README.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Optionally to enable Login to the dashboard
 ### Before you begin
 
 * You must enable basic auth to prevent user-functions from accessing the admin API of the gateway
-* A list of valid users is defined in the CUSTOMERS file in this GitHub repo, this acts as an ACL, but you can define your own
```
Comment (ewilde) [inline docs/README.md]: Valid users are github or gitlab usernames? If you are installing this yourself, presumably you only want your own github username to begin with?

File: docs/README.md [hunk 3]
```diff
@@ -61,7 +61,7 @@ The GitHub app will deliver webhooks to your OpenFaaS Cloud instance every time
 
 * With the router configured the URL should be like: `http://system.openfaas.cloud/github-event`
 
-* If you don't have the router configured remove the `system-` prefix from `system-github-event` in `stack.yml` and set the URL like: `http://my.openfaas.cloud/functions/github-event`
```
Comment (ewilde) [inline docs/README.md]: This needs to be configured in the `Webhook URL` field of the GitHub application?

File: docs/README.md [hunk 4]
```diff
@@ -61,7 +61,7 @@ The GitHub app will deliver webhooks to your OpenFaaS Cloud instance every time
 
 * With the router configured the URL should be like: `http://system.openfaas.cloud/github-event`
```
Comment (ewilde) [inline docs/README.md]: What is the `router`?

File: docs/README.md [hunk 5]
```diff
@@ -117,7 +117,7 @@ echo -n "$WEBHOOK_SECRET" | docker secret create github-webhook-secret -
 
 #### Create a secret for your GitHub App's private key
 
-Download the `.pem` file from the GitHub App page, then save it as a file named `private-key` with no extension.
```
Comment (ewilde) [inline docs/README.md]: Couldn't see how to download the private key once the GH app is created, may say use the private key which you downloaded when the application was created?

File: docs/README.md [hunk 6]
```diff
@@ -140,7 +140,7 @@ private_key_filename: my-private-key
 
 ### Setup your customer access control list (ACL)
 
-Edit `customers_url` in gateway_config.yml.
```
Comment (ewilde) [inline docs/README.md]: Maybe introduce each config file as it's referenced. Presume `gateway_config.yml` is the main configuration file for OpenFaaS cloud? and not related to the faas gateway in anyway

File: docs/README.md [hunk 7]
```diff
@@ -140,9 +140,9 @@ private_key_filename: my-private-key
...
+Edit `customers_url` in gateway_config.yml?
 
-Enter a list of GitHub usernames for your customers, these are case-sensitive.
```
Comment (ewilde) [inline docs/README.md]: Duplicate ? mentioned in more detail in `Before you begin` https://github.com/openfaas/openfaas-cloud/pull/357/files#diff-e70100059cc3a7376740bbe9751a082eL34

Suggest remove or mention the location of the file

File: docs/README.md [hunk 8]
```diff
@@ -158,7 +158,7 @@ You can edit `buildshiprun_limits.yml` to set the memory limit for your function
 
 ### Deploy your container builder
 
-You need to generate the ```~/.docker/config.json``` using the ```docker login``` command. 
+You need to generate the ```~/.docker/config.json``` using the ```docker login``` command. ?
```
Comment (ewilde) [inline docs/README.md]: Maybe introduction explaining why the need to remove the credential store attribute

File: docs/README.md [hunk 9]
```diff
@@ -249,7 +249,7 @@ cat $HOME/.docker/config.json | docker secret create registry-secret -
 Create of-builder and of-buildkit:
 
 ```
-./of-builder/deploy_swarm.sh
```
Comment (ewilde) [inline docs/README.md]: when I ran this command I got the following output:

```
./of-builder/deploy_swarm.sh
Error: No such container: of-buildkit
Unable to find image 'alexellis2/buildkit:2018-04-17' locally
2018-04-17: Pulling from alexellis2/buildkit
ff3a5c916c92: Already exists
41b99add21a2: Pull complete
75b79d1d2c70: Pull complete
c5da8e9d31f3: Pull complete
15eb2f52b88e: Pull complete
Digest: sha256:a05aad0a2344ee7b643a2ecbe5073c8636cee73b3c2f36ce13d8afe262311188
Status: Downloaded newer image for alexellis2/buildkit:2018-04-17
1330cd74d587ab40175f19944fb5bc37fa3567e33be70f46ae9d61bad1b49295
docker: Error response from daemon: linux runtime spec devices: lstat /dev/bpf23: input/output error.
o3dgh1arm1ki9uhlkbpwvyry9
```

Nothing in docker logs

File: docs/README.md (hunk 3 above)
Comment (ewilde) [reply docs/README.md]: Also doesn't explain what the `router` is and why you should have one / don't need one

File: docs/README.md (hunk 9 above)
Comment (ewilde) [reply docs/README.md]: installing docker edge fixed this problem, worth having a trouble shooting section at the end?

File: docs/README.md [hunk 10]
```diff
@@ -258,14 +258,13 @@ In gateway_config.yml
...
-  gateway_url: http://gateway.openfaas:8080/
-  gateway_public_url: http://of-cloud.public-facing-url.com:8080/
-  audit_url: http://gateway.openfaas:8080/function/audit-event
+  ...
   repository_url: docker.io/ofcommunity/
   push_repository_url: docker.io/ofcommunity/
+  ...
 ```
 
-Replace "ofcommunity" with your Docker Hub account i.e. `alexellis2/cloud/` or replace the whole string with the address of your private registry `reg.my-domain.xyz`.
```
Comment (ewilde) [inline docs/README.md]: Not clear here `alexellis2/cloud/` does not look like a Docker Hub account?

File: docs/README.md [hunk 11]
```diff
@@ -98,17 +98,17 @@ environment:
...
 WEBHOOK_SECRET="Long-Password-Phrase-Goes-Here"
 
 kubectl create secret generic -n openfaas-fn github-webhook-secret --from-literal github-webhook-secret="$WEBHOOK_SECRET"
 ```
 
-Swarm:
+##### Swarm:
 
 ```bash
 WEBHOOK_SECRET="Long-Password-Phrase-Goes-Here"
```
Comment (ewilde) [inline docs/README.md]: we don't mention any but presumably this secret also needs to be updated in the Github App  `Webhook secret (optional)` field?

**Comment by alexellis** [comment]
@Waterdrips please could you take a look over this feedback and submit anything you feel we may need to add into the README or documentation on ofc-bootstrap or the docs website?

**Pull request closed without merging**